docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,341 @@
+# ClusterClass
+
+The ClusterClass feature introduces a new way to create clusters which reduces boilerplate and enables flexible and powerful customization of clusters.
+ClusterClass is a powerful abstraction implemented on top of existing interfaces and offering a set of tools and operations to streamline cluster lifecycle management while maintaining the same underlying API.
+
+This tutorial covers using Cluster API and ClusterClass to create a Kubernetes cluster and how to perform a one-touch Kubernetes version upgrade.
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This guide covers creating a ClusterClass cluster with the CAPD provider.
+</aside>
+
+## Installation
+
+### Prerequisites
+
+#### Install tools
+
+This guide requires the following tools are installed: 
+- Install and setup [kubectl] and [clusterctl]
+- Install [Kind] and [Docker]
+
+For more information see the [quick start guide].
+
+#### Enable experimental features
+
+ClusterClass is currently behind a feature gate that needs to be enabled.
+This tutorial will also use another experimental gated feature - Cluster Resource Set - in this tutorial. This is not required for ClusterClass but is used in this tutorial to set up networking.
+
+To enable these features set the respective environmental variables by running:
```

Comment:
```suggestion
To enable these features set the respective environment variables by running:
```
nit (but just my preference, as that's what they are usually called in my experience)